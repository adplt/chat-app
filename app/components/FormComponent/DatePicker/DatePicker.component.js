import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {View, Text} from 'react-native';
import DatePickerView from './DatePickerView.component';
import styles from './DatePicker.styles';
import noop from 'lodash/noop';
import ErrorTextIndicator from '../../ErrorTextIndicatorComponent/ErrorTextIndicator.component';
import isEmpty from 'lodash/isEmpty';

export default class DatePickerComponent extends PureComponent {

  static propTypes = {
    disabled: PropTypes.bool,
    meta: PropTypes.object,
    placeholder: PropTypes.string,
    input: PropTypes.object,
    theme: PropTypes.string,
    mode: PropTypes.string,
    containerStyle: PropTypes.object,
    label: PropTypes.string,
    labelStyle: PropTypes.object,
    style: PropTypes.object,
    currentValue: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.string]),
    onValChange: PropTypes.func,
  };

  static defaultProps = {
    input: {onChange: noop, value: ''},
    style: {},
    mode: 'date',
    theme: 'primary',
    disabled: false,
    label: '',
    currentValue: new Date(),
    onValChange: noop,
  };

  state = {
    modalVisible: false
  };

  showModal = () => this.setState({modalVisible: true});

  hideModal = () => this.setState({modalVisible: false});

  onDateChange = (date) => {
    const {onValChange, input} = this.props;
    input.onChange(date);
    onValChange(date);
  };

  render () {
    const {
      meta,
      mode,
      input,
      placeholder,
      containerStyle,
      theme,
      disabled,
      label,
      labelStyle,
      style,
      currentValue
    } = this.props;
    const err = !disabled && (meta && meta.touched && !meta.active && meta.error);
    const wrapperStyle = disabled
      ? [styles.container[theme], styles.disabledInput]
      : styles.container[theme];
    return (
      <View style={style}>
        {
          !isEmpty(label) &&
            <Text style={[styles.label, labelStyle]}>{label}</Text>
        }
        <View style={[wrapperStyle, containerStyle]}>
          <DatePickerView
            selectedValue={input.value || currentValue}
            placeholder={placeholder}
            onDateChange={this.onDateChange}
            modalVisible={this.state.modalVisible}
            showModal={this.showModal}
            hideModal={this.hideModal}
            theme={theme}
            mode={mode}
            disabled={disabled}
          />
        </View>
        {err && <ErrorTextIndicator text={err}/>}
      </View>
    );
  }
}
