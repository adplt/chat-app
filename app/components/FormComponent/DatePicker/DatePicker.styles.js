import {theme} from '../../../styles/core.style';
import {fontSizeMedium, fontWeightMedium} from '../../../styles/common.style';

export default {
  container: {
    primary: {
      backgroundColor: theme.inputBackground,
      marginBottom: 10,
    },
    secondary: {
      backgroundColor: theme.transparent,
      borderBottomWidth: 1,
      borderBottomColor: theme.textInputBorder,
    }
  },
  iosPickerBackgroundColor: {
    backgroundColor: theme.white
  },
  label: {
    ...fontSizeMedium,
    ...fontWeightMedium,
    marginTop: 10,
    marginBottom: 5,
  },
  disabledInput: {
    backgroundColor: theme.disabledGrey,
  }
};
