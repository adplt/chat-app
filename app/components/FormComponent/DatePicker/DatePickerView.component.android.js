import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {DatePickerAndroid, TimePickerAndroid, View, Text} from 'react-native';
import Touchable from '../../TouchableComponent/Touchable.component';
import styles from './DatePickerView.styles';
import moment from 'moment';

export default class DatePickerView extends Component {

  static propTypes = {
    onDateChange: PropTypes.func,
    selectedValue: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
    placeholder: PropTypes.string,
    textPickerStyle: PropTypes.object,
    theme: PropTypes.string,
    mode: PropTypes.string,
    disabled: PropTypes.bool,
  };

  onDatePicked = ({action, year, month, day, hour, minute}) => {
    const {mode, selectedValue} = this.props;
    if (mode === 'date' || mode === 'datetime') {
      if (action !== DatePickerAndroid.dismissedAction) {
        const hourTime = moment(selectedValue).hour();
        const minuteTime = moment(selectedValue).minute();
        this.props.onDateChange(moment({year, month, day, hour: hourTime, minute: minuteTime}).toDate());
      }
    } else {
      if (action !== TimePickerAndroid.dismissedAction) {
        const yearTime = moment(selectedValue).year();
        const monthTime = moment(selectedValue).month();
        const dayTime = moment(selectedValue).date();
        this.props.onDateChange(moment({year: yearTime, month: monthTime, day: dayTime, hour, minute}).toDate());
      }
    }
  };

  showPicker = () => {
    const {mode} = this.props;
    if (mode === 'date' || mode === 'datetime') {
      DatePickerAndroid.open({
        date: new Date(moment(this.props.selectedValue).format()),
      }).then(this.onDatePicked);
    } else {
      TimePickerAndroid.open({
        is24Hour: true,
        hour: moment(this.props.selectedValue).hour(),
        minute: moment(this.props.selectedValue).minute(),
      }).then(this.onDatePicked);
    }
  };

  render () {
    const {theme = 'primary', selectedValue, placeholder, textPickerStyle, mode, disabled} = this.props;
    let timeFormat = 'DD MMM YYYY';
    if (mode === 'time') {
      timeFormat = 'kk:mm';
    }

    return (
      <View>
        <Touchable onPress={this.showPicker} style={styles.textBoxStyle} disabled={disabled}>
          {
            selectedValue ?
              <Text style={[styles.input[theme], textPickerStyle]}>{moment(selectedValue).format(timeFormat)}</Text> :
              <Text style={[styles.input[theme], styles.placeholder[theme]]}>{placeholder}</Text>
          }
        </Touchable>
      </View>
    );
  }
}
