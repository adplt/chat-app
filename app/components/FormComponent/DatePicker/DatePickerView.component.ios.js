import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {DatePickerIOS, View, Text} from 'react-native';
import Touchable from '../../TouchableComponent/Touchable.component';
import OverlayModal from '../../OverlayModalComponent/OverlayModal.component';
import styles from './DatePickerView.styles';
import moment from 'moment';
import {language} from '../../../config/language';

export default class DatePickerView extends Component {

  static propTypes = {
    selectedValue: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.string]),
    textPickerStyle: PropTypes.object,
    onDateChange: PropTypes.func,
    showModal: PropTypes.func,
    hideModal: PropTypes.func,
    modalVisible: PropTypes.bool,
    disabled: PropTypes.bool,
    placeholder: PropTypes.string,
    theme: PropTypes.string,
    mode: PropTypes.string,
  };

  render () {
    const {
      theme,
      mode,
      selectedValue,
      onDateChange,
      showModal,
      hideModal,
      modalVisible,
      textPickerStyle,
      placeholder,
      disabled
    } = this.props;
    let timeFormat = 'DD MMM YYYY';
    if (mode === 'time') {
      timeFormat = 'kk:mm';
    } else if (mode === 'datetime') {
      timeFormat = 'DD MMM YYYY kk:mm';
    }

    return (
      <View>
        <Touchable onPress={showModal} style={styles.textBoxStyle} disabled={disabled}>
          {
            selectedValue ?
              <Text style={[styles.input[theme], textPickerStyle]}>{moment(selectedValue).format(timeFormat)}</Text> :
              <Text style={[styles.input[theme], styles.placeholder[theme]]}>{placeholder}</Text>
          }
        </Touchable>
        <OverlayModal
          animationType='fade'
          visible={modalVisible}
          containerStyles={styles.overlayContainer}
          innerContainerStyles={styles.innerOverlay}
          onClose={hideModal}
          closeOnTouchOutside
        >
          <View style={styles.modalContainer}>
            <View style={styles.buttonContainer}>
              <Touchable onPress={hideModal}>
                <Text style={styles.buttonStyle}>
                  {language.DATE_PICKER__DONE}
                </Text>
              </Touchable>
            </View>

            <View>
              <DatePickerIOS
                style={styles.bottomPicker}
                date={new Date(moment(selectedValue).format())}
                mode={mode}
                onDateChange={onDateChange}
              />
            </View>
          </View>
        </OverlayModal>
      </View>
    );
  }
}
