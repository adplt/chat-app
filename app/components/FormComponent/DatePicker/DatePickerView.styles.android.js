import {fontSizeMedium, fontSizeSmall, textPrimayColor} from '../../../styles/common.styles';
import {theme} from '../../../styles/core.styles';

export default {
  input: {
    primary: {
      ...fontSizeMedium,
      padding: 10,
    },
    secondary: {
      ...fontSizeMedium,
      ...textPrimayColor,
      paddingVertical: 10,
    }
  },

  textBoxStyle: {
    height: 50,
    justifyContent: 'center',
  },

  textStyle: {
    ...fontSizeSmall,
  },
  placeholder: {
    primary: {
      ...fontSizeSmall,
      color: theme.placeholderTextColor,
    },
    secondary: {
      ...fontSizeMedium,
      color: theme.placeholderTextColor,
    },
  },
};
