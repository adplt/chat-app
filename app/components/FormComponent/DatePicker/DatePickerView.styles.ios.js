import {theme} from '../../../styles/core.style';
import {viewPortWidth} from '../../../utils/device.util';
import {
  fontWeightBold,
  fontSizeMedium,
  textPrimayColor
} from '../../../styles/common.style';

export default {
  input: {
    primary: {
      ...fontSizeMedium,
      padding: 10,
    },
    secondary: {
      ...fontSizeMedium,
      ...textPrimayColor,
      paddingVertical: 10,
    },
  },
  overlayContainer: {
    justifyContent: 'flex-end',
    padding: 0,
  },
  innerOverlay: {
    backgroundColor: 'transparent',
    padding: 0,
  },
  textBoxStyle: {
    height: 50,
    justifyContent: 'center',
  },
  modalContainer: {
    width: viewPortWidth,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 0,
    backgroundColor: theme.inputBackground,
  },
  buttonContainer: {
    width: viewPortWidth,
    padding: 20,
    height: 50,
    alignItems: 'center',
    justifyContent: 'flex-end',
    flexDirection: 'row',
  },
  bottomPicker: {
    width: viewPortWidth,
    backgroundColor: theme.white,
  },
  buttonStyle: {
    ...fontSizeMedium,
    ...fontWeightBold,
    color: theme.buttonPickerColor,
  },
  placeholder: {
    primary: {
      ...fontSizeMedium,
      color: theme.placeholderTextColor,
    },
    secondary: {
      ...fontSizeMedium,
      color: theme.placeholderTextColor,
    },
  },
};
