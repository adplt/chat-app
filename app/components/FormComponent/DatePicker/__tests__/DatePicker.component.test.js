import React from 'react';
import renderer from 'react-test-renderer';
import DatePicker from '../DatePicker.component';
import noop from 'lodash/noop';

describe('DatePicker component', () => {
  it('renders correctly', () => {
    const input = {onChange: noop, value: new Date('2000-01-01')};
    const tree = renderer.create(<DatePicker input={input}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
