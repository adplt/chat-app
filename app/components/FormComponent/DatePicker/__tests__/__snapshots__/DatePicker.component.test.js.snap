// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker component renders correctly 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#f0f1f1",
          "marginBottom": 10,
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 19,
                "fontWeight": "500",
                "padding": 10,
              },
              undefined,
            ]
          }
        >
          01 Jan 2000
        </Text>
      </View>
      <Modal
        animationOut="fadeOut"
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <TouchableWithoutFeedback
            onPress={[Function]}
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0.75)",
                    "flex": 1,
                    "justifyContent": "center",
                    "padding": 20,
                  },
                  Object {
                    "justifyContent": "flex-end",
                    "padding": 0,
                  },
                ]
              }
            >
              <TouchableWithoutFeedback
                onPress={[Function]}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "white",
                        "padding": 20,
                      },
                      Object {
                        "backgroundColor": "transparent",
                        "padding": 0,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#f0f1f1",
                        "justifyContent": "center",
                        "padding": 0,
                        "width": 750,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "height": 50,
                          "justifyContent": "flex-end",
                          "padding": 20,
                          "width": 750,
                        }
                      }
                    >
                      <View
                        accessibilityComponentType={undefined}
                        accessibilityLabel={undefined}
                        accessibilityTraits={undefined}
                        accessible={true}
                        collapsable={undefined}
                        hitSlop={undefined}
                        isTVSelectable={true}
                        nativeID={undefined}
                        onLayout={undefined}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                        testID={undefined}
                        tvParallaxProperties={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#007AFF",
                              "fontSize": 19,
                              "fontWeight": "800",
                            }
                          }
                        >
                          Selesai
                        </Text>
                      </View>
                    </View>
                    <View>
                      <DatePickerIOS
                        date={2000-01-01T00:00:00.000Z}
                        mode="date"
                        onDateChange={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "width": 750,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </TouchableWithoutFeedback>
            </View>
          </TouchableWithoutFeedback>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;
